pkg/managerdriver/translate_ingresses.go

```diff
@@ -257,19 +256,17 @@ func (t *translator) ingressBackendToIR(ingress *netv1.Ingress, backend *netv1.I
 			return nil, fmt.Errorf("unable to resolve traffic policy backend for ingress rule: %w", err)
 		}
 
-		var routeTrafficPolicy trafficpolicy.TrafficPolicy
-		if len(routePolicyCfg.Spec.Policy) > 0 {
-			if err := json.Unmarshal(routePolicyCfg.Spec.Policy, &routeTrafficPolicy); err != nil {
-				return nil, fmt.Errorf("failed to unmarshal traffic policy: %w. raw traffic policy: %v", err, routePolicyCfg.Spec.Policy)
-			}
+		routeTrafficPolicy, err := trafficpolicy.NewTrafficPolicyFromJSON(routePolicyCfg.Spec.Policy)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal traffic policy: %w. raw traffic policy: %v", err, routePolicyCfg.Spec.Policy)
```

Comment:
```suggestion
			return nil, err
```

I think we get most of this error message from the `NewTrafficPolicyFromJSON` function. Wrapping seems like it'll just be overly verbose here. My suggestions is to just return this error here unless we add additional context.